metrics/resource.go

```diff
@@ -15,11 +16,28 @@ type Resource struct {
 	PodMetadata *PodMetadata
 }
 
-func getNodeResource(s stats.NodeStats) *Resource {
+func getNodeResource(s stats.NodeStats, metadata *Metadata) *Resource {
 	labels := map[string]string{
 		LabelNodeName: s.NodeName,
 	}
 
+	if metadata.IncludeNodeLabels {
+		nodeMetadata, err := metadata.GetNodeMetadataByName(s.NodeName)
+		if err != nil {
+			return &Resource{
+				Type:      "node",
+				Name:      s.NodeName,
+				Labels:    labels,
+				Timestamp: s.CPU.Time.Time,
+			}
+		}
+
+		nodeLabels := nodeMetadata.GetLabels()
+		for k, v := range nodeLabels {
+			labels[PrefixLabel+k] = v
+		}
```

Comment:
I might invert the normal early return here so you don't duplicate the Resource definition:

```suggestion
		if err == nil {
			for k, v := range nodeMetadata.GetLabels() {
				labels[PrefixLabel+k] = v
			}
		}
```